extensions/HyperVExtension/src/HyperVExtension/Models/HyperVVirtualMachine.cs

```diff
@@ -223,7 +253,7 @@ public IAsyncOperation<ComputeSystemOperationResult> ShutDownAsync(string option
             }
             catch (Exception ex)
             {
-                StateChanged(this, ComputeSystemState.Unknown);
+                StateChanged(this, GetState());

```

Comment:
[preexisting] There is a danger making OOP (or other complex code) here under the “catch”. If an exception happens the extension will likely crash. And if not crash it will never send operation results back. We should refactor this (later) to handle possible exception and always return ComputeSystemOperationResult.

Reply:
fair, I'll follow up with you on this after this